packages/core/src/navigation-item/NavigationItem.tsx

```diff
@@ -1,14 +1,60 @@
+import React, {
+  AriaAttributes,
+  ForwardedRef,
+  MouseEvent,
+  ReactElement,
+} from "react";
 import { ComponentPropsWithoutRef, forwardRef, MouseEventHandler } from "react";
 import { makePrefixer } from "../utils";
 import { clsx } from "clsx";
 import { ExpansionIcon } from "./ExpansionIcon";
-import { ConditionalWrapper } from "./ConditionalWrapper";
 
 import { useWindow } from "@salt-ds/window";
 import { useComponentCssInjection } from "@salt-ds/styles";
 
 import navigationItemCss from "./NavigationItem.css";
 
+type OptionalPartial<T, K extends keyof T> = Partial<Pick<T, K>>;
+
+export interface NavigationItemElementProps<
+  T extends HTMLAnchorElement | HTMLButtonElement
+> extends Pick<
+    AriaAttributes,
+    "aria-label" | "aria-expanded" | "aria-current"
+  > {
+  /**
+   * Item's children.
+   */
+  children: ReactElement;
+  /**
+   * Item's className.
+   */
+  className: string;
+  /**
+   * Handler to expand groups.
+   */
+  onClick?: MouseEventHandler<T>;
+}
+
+export interface NavigationItemRenderProps<
+  T extends HTMLAnchorElement | HTMLButtonElement
+> extends OptionalPartial<
+    NavigationItemProps,
+    "active" | "expanded" | "level" | "orientation"
+  > {
+  /**
+   * If the item to render is expandable when clicked.
+   */
+  isParent: boolean;
```

Comment:
Should this and `href` be picked from `NavigationItemProps`'s `parent` and `href` as well?

Reply:
A Navigation item is either a `button` or a `anchor`, depending on `parent`, updated the docs to make this clearer.